src/rest.jl

```diff
@@ -184,5 +184,11 @@ function request(
     headers = _ensure_headers!(headers)
     _authenticate!(ctx, headers)
     opts = (;redirect = false, connection_limit = 4096)
-    return HTTP.request(method, url, headers; query = query, body = body, opts..., kw...)
```

Comment:
Yeah, this isn't quite the right way to go about it. You _can_ pass `status_exception=false`, but I don't recommend it since it prevents retries and some other oddities. I think the first question is whether we want something like this in the SDK itself because we think we'll _always_ want it, no matter who is using/what application we're using it in; or whether this would be better from the external application/script where we're calling the SDK method.

Once we answer that question, I think a better implementation would be something like:

```julia
try
    # if we keep this inside the SDK
    return HTTP.request(...)
catch e
    if e isa HTTP.StatusError
        @warn e.response
        # log other stuff
    end
    rethrow()
end
```

Reply:
thanks @bachdavi you are right
```
                                                                                                                                       
julia> HTTP.request("GET", "https://httpbin.org/status/400")                                                                                                                                                                                                                   ERROR: HTTP.Exceptions.StatusError(400, "GET", "/status/400", HTTP.Messages.Response:                                                                                                                                                                                          """                                                                                                                                                                                                                                                                            HTTP/1.1 400 Bad Request                                                                                                                                                                                                                                                       Date:Wed, 14 Dec 2022 00:57:45 GMT                              
Content-Type: text/html; charset=utf-8                 
Content-Length: 0                                                                                                                                                                                                                                                              
Connection: keep-alive                                 
Server: gunicorn/19.9.0
Access-Control-Allow-Origin: *                              
Access-Control-Allow-Credentials: true

```
Closing this PR as the behavior is already implemented within the HTTP library